From 6f6f50c24a0a8e78b704761a6c3c612b57899005 Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@dell.com>
Date: Fri, 31 Jan 2020 11:07:39 -0600
Subject: [PATCH] trivial: fix plural translation (Fixes: #1732)

---
 src/fu-engine-helper.c | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/src/fu-engine-helper.c b/src/fu-engine-helper.c
index 4d403d0c..6260a10c 100644
--- a/src/fu-engine-helper.c
+++ b/src/fu-engine-helper.c
@@ -64,15 +64,10 @@ fu_engine_update_motd (FuEngine *self, GError **error)
 	}
 
 	str = g_string_new ("");
-	if (upgrade_count == 1) {
-		g_string_append_printf (str,
-					_("%u device has a firmware upgrade available."),
-					upgrade_count);
-	} else {
-		g_string_append_printf (str,
-					_("%u devices have a firmware upgrade available."),
-					upgrade_count);
-	}
+	g_string_append_printf (str, ngettext ("%u device has a firmware upgrade available.",
+					       "%u devices have a firmware upgrade available.",
+					       upgrade_count),
+					       upgrade_count);
 	g_string_append_printf (str,
 				"\n%s\n",
 				_("Run `fwupdmgr get-upgrades` for more information."));
-- 
2.24.0

