From 5c7e17276c8eb0c2a845cecb7d644a4bccf80519 Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario.limonciello@dell.com>
Date: Thu, 13 Feb 2020 12:06:41 -0600
Subject: [PATCH] fu-util: fix a logic error in report uploading introduced in
 e076d48afb74d1447936d353f12781755f13b047

Fixes: #1770
---
 src/fu-util.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/fu-util.c b/src/fu-util.c
index 5bcffede..e6531c90 100644
--- a/src/fu-util.c
+++ b/src/fu-util.c
@@ -286,16 +286,12 @@ fu_util_maybe_enable_automatic (FuUtilPrivate *priv, GPtrArray *remotes, GError
 							 NULL, error))
 				return FALSE;
 		}
-		return TRUE;
+		break;
 	default:
 		break;
 	}
 
-	g_set_error_literal (error,
-			     FWUPD_ERROR,
-			     FWUPD_ERROR_NOTHING_TO_DO,
-			     "invalid option selected");
-	return FALSE;
+	return TRUE;
 }
 
 static gboolean
-- 
2.20.1

