From bb49bd52b9a4f9c829c26e09741fcbcf10c6a984 Mon Sep 17 00:00:00 2001
From: Mario Limonciello <mario_limonciello@dell.com>
Date: Mon, 21 Mar 2016 08:56:52 -0500
Subject: [PATCH] Add Alienware to the version quirk table

---
 src/fu-quirks.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/fu-quirks.h b/src/fu-quirks.h
index 6f378a9..d91824b 100644
--- a/src/fu-quirks.h
+++ b/src/fu-quirks.h
@@ -29,8 +29,9 @@ typedef struct {
 } FuVendorQuirks;
 
 static const FuVendorQuirks quirk_table[] = {
-	/* Dell uses AA.BB.CC.DD rather than AA.BB.CCDD */
+	/* Dell & Alienware use AA.BB.CC.DD rather than AA.BB.CCDD */
 	{ "Dell Inc.",	"com.dell.uefi",	AS_VERSION_PARSE_FLAG_NONE },
+	{ "Alienware",	"com.dell.uefi",	AS_VERSION_PARSE_FLAG_NONE },
 	{ NULL,		NULL,			AS_VERSION_PARSE_FLAG_NONE }
 };
 
-- 
1.9.1

