From d5988a533948e1912c50e93ab27183def7319d92 Mon Sep 17 00:00:00 2001
From: Ken VanDine <ken.vandine@canonical.com>
Date: Tue, 12 Nov 2019 09:36:39 -0800
Subject: [PATCH] Allows confined snaps to activate fwupd via D-Bus

---
 data/org.freedesktop.fwupd.service.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/data/org.freedesktop.fwupd.service.in b/data/org.freedesktop.fwupd.service.in
index af051972..53f51710 100644
--- a/data/org.freedesktop.fwupd.service.in
+++ b/data/org.freedesktop.fwupd.service.in
@@ -4,3 +4,4 @@ Documentation=https://fwupd.org/
 Exec=@libexecdir@/fwupd/fwupd
 User=root
 SystemdService=fwupd.service
+AssumedAppArmorLabel=unconfined
-- 
2.20.1

